### ISSUE TITLE
Run all chunks above should ignore chunks with `error = TRUE`

### ISSUE THREAD

**Issue by hadley** [rstudio/rstudio#11411]
### System details

    RStudio Edition : 
    RStudio Version : 
    OS Version      : 
    R Version       : 

### Steps to reproduce the problem

Create an Rmd with the following content:

````
```{r, error = TRUE}
1 + "x"
```

```{r error = TRUE}
#| error: true
2 + "x"
```

```{r}
3
```

```{r}
4
```
````

Click the "run all chunks" above in the chunk 4.

### Describe the problem in detail

Chunk 3 is never executed because chunk 1 errors, even though this is what I expected (since I've labelled it with `error = TRUE`) and Rmarkdown can still run the whole doc.

### Describe the behavior you expected

I _think_ I expect chunks with `error = TRUE` to either be skipped or (probably better) run and not terminate chunk execution (i.e. as if they had been wrapped in a `try()`)

- [x] I have read the guide for [submitting good bug reports](https://github.com/rstudio/rstudio/wiki/Writing-Good-Bug-Reports).
- [x] I have installed the latest version of RStudio, and confirmed that the issue still persists.
- [x] If I am reporting an RStudio crash, I have included a [diagnostics report](https://support.rstudio.com/hc/en-us/articles/200321257-Running-a-Diagnostics-Report).
- [x] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

**Comment by jgutman** [comment]
See also #9436, which handled a regression with warning and message chunk options. At that time, the error chunk option was tested as well and was working correctly.

See also #10645, which added handling for the YAML style chunk options `#| error: true`

Both of these fixes were added in 2022.06.0

**Comment by jgutman** [comment]
There's a small error in the reprex above, in that it is missing the comma in `{r error = TRUE}` 

Changing the reprex to 

~~~~
```{r, error = TRUE}
1 + "x"
```

```{r}
#| error: true
2 + "x"
```

```{r}
3
```

```{r}
4
```
~~~~

And clicking, Run All Chunks Above on chunk 4, chunk 1 executes, displaying an error message, chunk 2 executes, displaying an error message, and chunk 3 executes successfully, which is expected behavior

**Comment by jgutman** [comment]
> Describe the behavior you expected
> I think I expect chunks with error = TRUE to either be skipped or (probably better) run and not terminate chunk execution (i.e. as if they had been wrapped in a try())

Yes, the latter is the expected behavior (and what I'm observing in the latest build)

**Comment by hadley** [comment]
Oh, as usual, this is due to my use of "chunk output in console".
